src/reducers/graphReducer.ts

```diff
@@ -5,16 +5,18 @@ import _ from 'lodash';
 import { defaultNodeLabel, EdgeData, NodeData } from "../logics/utils";
 
 type GraphState = {
-  nodes: NodeData[];
-  edges: EdgeData[];
+  nodes: Node[];
+  edges: Edge[];
```

Comment:
These should be NodeData and EdgeData. The definitions are nearly the same, but we should manage our own Node/Edge abstractions instead of relying on Vis.